yarn-project/simulator/src/private/providers/acvm_native.ts

```diff
@@ -136,12 +138,21 @@ export async function executeNativeCircuit(
 
 export class NativeACVMSimulator implements SimulationProvider {
   constructor(private workingDirectory: string, private pathToAcvm: string, private witnessFilename?: string) {}
-  async executeProtocolCircuit(input: WitnessMap, compiledCircuit: NoirCompiledCircuit): Promise<WitnessMap> {
+
+  async executeProtocolCircuit(
+    input: ACVMWitness,
+    artifact: NoirCompiledCircuitWithName,
+    callback: ForeignCallHandler | undefined,
+  ): Promise<ACVMWitness> {
     // Execute the circuit on those initial witness values
 
+    if (callback) {
+      logger.warn('Native ACVM simulator does not support foreign calls. Ignoring callback.');
+    }
```

Comment:
I'd consider failing here - this warning will get lost in the logs, and it seems like if we ever run into this it'd be very annoying for whomever did it.

Reply:
Makes sense. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/12148/commits/9f80fc06ac499237516a226384c0ee384780ff8f